mubench.pipeline/tasks/implementations/collect_versions.py

```diff
@@ -0,0 +1,10 @@
+from data.project import Project
+from utils.data_filter import DataFilter
+
+
+class CollectVersions:
+    def __init__(self, data_filter: DataFilter):
+        self.data_filter = data_filter
+
+    def run(self, project: Project):
+        return [version for version in project.versions if not self.data_filter.is_filtered(version.id)]
```

Comment:
Same as with `version.misuses`.

Reply:
`project.versions` is indeed only used when uploading metadata (to check whether a misuse is referenced from multiple versions). We could probably replace this usage. We only do this to raise an error, because we currently cannot handle misuses referenced from multiple versions at that point.